src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,39 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> list = List.of();
+        if (type.equals(Bulldozer.class)) {
+            list = new BulldozerProducer().get();
+        }
+        if (type.equals(Truck.class)) {
+            list = new TruckProducer().get();
+        }
+        if (type.equals(Excavator.class)) {
+            list = new ExcavatorProducer().get();
+        }
+        return new ArrayList<>(list);
+    }
+
+    @Override
+    public void startWorking(List<? extends Machine> list) {
+        for (int i = 0; i < list.size(); i++) {
+            list.get(i).doWork();
+        }
+    }
```

Comment:
As an option you can use for-each loop to avoid size() and get(i) methods invocation.